server/routerlicious/packages/lambdas/src/scribe/checkpointManager.ts

```diff
@@ -136,19 +140,62 @@ export class CheckpointManager implements ICheckpointManager {
 		}
 	}
 
-	private async writeScribeCheckpointState(checkpoint: IScribe) {
-		await this.documentRepository.updateOne(
-			{
-				documentId: this.documentId,
-				tenantId: this.tenantId,
-			},
-			{
-				// MongoDB is particular about the format of stored JSON data. For this reason we store stringified
-				// given some data is user generated.
-				scribe: JSON.stringify(checkpoint),
-			},
-			null,
-		);
+	private async writeScribeCheckpointState(checkpoint: IScribe, noActiveClients: boolean) {
+		const lumberProperties = getLumberBaseProperties(this.documentId, this.tenantId);
+
+		if (!this.localCheckpointEnabled || !this.checkpointRepository) {
+			// Write to global collection when local checkpoints are disabled or there is no checkpoint repository
+			await this.writeScribeCheckpointToCollection(checkpoint, false);
+			return;
+		}
+
+		await (!noActiveClients
+			? this.writeScribeCheckpointToCollection(checkpoint, true)
+			: Promise.all([
+					this.checkpointRepository
+						.deleteCheckpoint(this.documentId, this.tenantId)
+						.catch((error) => {
+							Lumberjack.error(
+								`Error removing checkpoint data from the local database.`,
+								lumberProperties,
+								error,
+							);
```

Comment:
If delete local failed, we are OK to continue write to global, right? Just double confirm